### ISSUE TITLE
Failed to restart the kim builder on macos

### ISSUE THREAD

**Issue by codering** [rancher-sandbox/rancher-desktop#1150]
### Rancher Desktop Version

0.7.1

### Rancher Desktop K8s Version

1.22.5

### What operating system are you using?

macOS

### Operating System / Build Version

11.0.1 

### What CPU architecture are you using?

x64

### Actual Behavior

```
2021-12-22T02:15:52.973Z: Installing kim: kim builder install --force --endpoint-addr 127.0.0.1
2021-12-22T02:17:17.157Z: Failed to restart the kim builder: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/kim exited with code 1.
2021-12-22T02:17:17.157Z: Attempt to run 'kim install builder' => error: 'time="2021-12-22T10:15:53+08:00" level=info msg="Applying node-role `builder` to `lima-rancher-desktop`"
time="2021-12-22T10:15:55+08:00" level=info msg="Asserting namespace `kube-image`"
time="2021-12-22T10:15:55+08:00" level=info msg="Asserting TLS secrets"
time="2021-12-22T10:15:56+08:00" level=info msg="Asserting service/endpoints"
time="2021-12-22T10:15:56+08:00" level=info msg="Installing builder daemon"
time="2021-12-22T10:15:56+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:02+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:08+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:15+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:20+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:25+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:32+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:37+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:42+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:47+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:16:54+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:17:00+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:17:05+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:17:11+08:00" level=info msg="Waiting on builder daemon availability..."
time="2021-12-22T10:17:17+08:00" level=info msg="Waiting on builder daemon availability..."
Error: timeout waiting for builder to become available
Usage:
  kim builder install [OPTIONS]

Flags:
      --agent-image string         Image to run the agent w/ missing tag inferred from version
      --agent-port int             Port that the agent will listen on (default 1233)
      --buildkit-image string      BuildKit image for running buildkitd (default "docker.io/moby/buildkit:v0.8.3")
      --buildkit-port int          BuildKit service port (default 1234)
      --buildkit-socket string     BuildKit socket address (default "unix:///run/buildkit/buildkitd.sock")
      --containerd-socket string   Containerd socket address (default on k3s "/run/k3s/containerd/containerd.sock")
      --containerd-volume string   Containerd storage volume (default on k3s "/var/lib/rancher")
      --force                      Force installation by deleting existing builder
  -h, --help                       help for install
      --no-fail                    Do not fail if backend components are already installed
      --no-wait                    Do not wait for backend to become available
      --selector string            Selector for nodes (label query) to apply builder role

Global Flags:
  -x, --context string      kubeconfig context for authentication
      --debug               
      --debug-level int     
  -k, --kubeconfig string   kubeconfig for authentication
  -n, --namespace string    namespace (default "kube-image")

time="2021-12-22T10:17:17+08:00" level=fatal msg="timeout waiting for builder to become available"
'
2021-12-22T02:17:17.157Z: A reset might be necessary to support building images.
```

### Steps to Reproduce

start  after installed.

### Result

![image](https://user-images.githubusercontent.com/2271825/147024689-43d592ce-58f3-4184-9910-4ff239cd4e56.png)

### Expected Behavior

No error.

**Comment by Crayeth** [comment]
Same issue occurred.

MacOs 12.1
MacBook Pro (15-inch, 2019)
Processor 2,3 GHz 8-Core Intel Core i9
Rancher Desktop 0.7.1
K8s 1.21.7 (Stable)
```
2021-12-23T08:10:28.642Z: Existing kim install invalid: incorrect endpoint address.
2021-12-23T08:10:28.643Z: Installing kim: kim builder install --force --endpoint-addr 127.0.0.1
2021-12-23T08:11:50.625Z: Failed to restart the kim builder: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/kim exited with code 1.
2021-12-23T08:11:50.626Z: Attempt to run 'kim install builder' => error: 'time="2021-12-23T09:10:29+01:00" level=info msg="Applying node-role `builder` to `lima-rancher-desktop`"
```
...
```
Error: timeout waiting for builder to become available
```
...
```
time="2021-12-23T09:11:50+01:00" level=fatal msg="timeout waiting for builder to become available"
'
2021-12-23T08:11:50.626Z: A reset might be necessary to support building images.
```

**Comment by evertonlperes** [comment]
I've seen this issue in a really slow environment (e.g.: virtualbox).
Could you check if after the error shows up, the builder pod will be available (up and running without issues)? 
You can check it running: `watch -n 1 "kubectl get pods -A"`

**Comment by Crayeth** [comment]
> I've seen this issue in a really slow environment (e.g.: virtualbox). Could you check if after the error shows up, the builder pod will be available (up and running without issues)? You can check it running: `watch -n 1 "kubectl get pods -A"`

I've given my environment 16gb of ram and 6 cpu on my macbook pro i9. I get the error after (re)starting. The builder seems to crash but does recover on its own without me pressing OK in the prompt:
```
NAMESPACE     NAME                                      READY   STATUS      RESTARTS   AGE
kube-system   helm-install-traefik-crd-8bfmn            0/1     Completed   0          2d
kube-system   helm-install-traefik-lnklf                0/1     Completed   1          2d
kube-system   local-path-provisioner-5ff76fc89d-gvgxs   1/1     Running     1          2d
kube-system   metrics-server-86cbb8457f-vmqr7           1/1     Running     1          2d
kube-system   coredns-7448499f4d-c6w48                  1/1     Running     1          2d
kube-system   svclb-traefik-n4998                       2/2     Running     2          2d
kube-system   traefik-6b84f7cbc-rzvvk                   1/1     Running     1          2d
kube-image    builder-l7krg                             2/2     Running     0          119s
```

**Comment by drenze-athene** [comment]
I'm experiencing this issue as well, when I start in a regular user account. However, when I start it from an account with admin access, it starts up as expected. I suspect the issue may be a permissions issue with `/private/var/run/rancher-desktop-lima`.

Additionally, I only get this issue if I first install `0.6.1`, then upgrade to `0.7.1`. If I do a clean install, directly to `0.7.1`, I get an `ECONNRESET` error while waiting for the kubernetes api to become available.

**Comment by jandubois** [comment]
> I'm experiencing this issue as well, when I start in a regular user account.

This is a separate issue, and as you noted in #1127, RD does not currently work as a regular (non-admin) user. We obviously need to fix this.

As I wrote in the other issue, `kim` is being removed from RD in the next release and buildkitd will run as a native service.
